packages/bvaughn-architecture-demo/src/suspense/createGenericCache.ts

```diff
@@ -18,7 +18,7 @@ interface HookState<TValue> {
 }
 
 export function createGenericCache<TParams extends Array<any>, TValue>(
-  fetchValue: (...args: TParams) => Promise<TValue>,
+  fetchValue: (...args: TParams) => Promise<TValue> | TValue,
```

Comment:
@hbenl I don't think there's any reason that a Suspense cache's underlying function _has_ to be async, right?

Reply:
Maybe we could do something special in this case and just avoid throwing on cache misses entirely?